### PR TITLE
MAT-9834: add Caffeine caching to CqlLibraryService

### PR DESCRIPTION
**Pull request by etan-sb**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-9834](https://jira.cms.gov/browse/MAT-9834)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java [hunk 1]
```diff
@@ -40,6 +45,15 @@ public void setUpLibrarySourceProvider(String cql, String accessToken) {
   }
 
   public String getLibraryCql(String name, String version, String accessToken) {
+    if (cacheManager != null) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java:48]: Why are we manually handling the cache instead of #Cacheable ?

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java:48]: I do not see a cache eviction policy either, is it not required here ?

**Review by RohitKandimalla — APPROVED** [review]
